olive/model/__init__.py

```diff
@@ -68,19 +67,25 @@ def __init__(
         self.io_config = None
         # store resource paths
         self.resource_paths: Dict[str, ResourcePath] = {}
-        # this is for storing local instances of resource paths
-        self.local_resource_paths: Dict[str, ResourcePath] = {}
-        resources = resources or {}
+        resources = {}
         resources["model_path"] = model_path
-        for resource_name, resource_path in resources.items():
-            self.resource_paths[resource_name] = create_resource_path(resource_path)
-            # initialize local resource paths to None
-            self.local_resource_paths[resource_name] = None
+        self.set_resources(resources)
 
     @property
     def model_path(self) -> str:
         """Return local model path."""
-        return self.get_local_resource("model_path")
+        return self.get_resource("model_path")
+
+    def set_resources(self, resources: Dict[str, OLIVE_RESOURCE_ANNOTATIONS]):
+        for resource_name, resource_path in resources.items():
+            if resource_path is not None:
```

Comment:
The logic looks duplicate with `self.set_resource`. Can we simplify as:
```python
for resource_name, resource_path in resource.items():
    self.set_resource(resouce_name, resource_path)
```

Reply:
actually, they are not same, the set_resource has validation to make sure the resource_name exist before update.

Reply:
Maybe it can be renamed to `validate_resources` to avoid confusion?

Reply:
maybe we can use a different name like `add_resources` or something to avoid confusion? Since `set_resource` assumes the resource name is already there?